src/pages/IPFSPin.vue

```diff
@@ -350,6 +371,70 @@ export default {
           path: this.cid_name
         })
       }
+    },
+    async putContent (message, content, inline_requested, api_server) {
+      let inline = inline_requested
+      if (inline) {
+        let serialized = JSON.stringify(content)
+        if (serialized.length > 150000) {
+          inline = false
+        } else {
+          message.item_type = 'inline'
+          message.item_content = serialized
+          // eslint-disable-next-line no-use-before-define
+          message.item_hash = new shajs.sha256().update(serialized).digest('hex') // eslint-disable-line no-use-before-define
+        }
+      }
+    },
+    async send (message) {
+      if (this.account.type === 'ETH') {
+        message = await ethereum.sign(this.account, message)
+      } else if (this.account.type === 'SOL') {
+        message = await solana.sign(this.account, message)
+      }
```

Comment:
Could this could be abstracted in the store, so you would always use the right signer instead of having a conditional switch for every message that you create?